superset-frontend/src/SqlLab/components/TemplateParamsEditor.tsx

```diff
@@ -26,34 +25,32 @@ import { debounce } from 'lodash';
 import ModalTrigger from 'src/components/ModalTrigger';
 import { ConfigEditor } from 'src/components/AsyncAceEditor';
 import { FAST_DEBOUNCE } from 'src/constants';
-
-const propTypes = {
-  onChange: PropTypes.func,
-  code: PropTypes.string,
-  language: PropTypes.oneOf(['yaml', 'json']),
-};
-
-const defaultProps = {
-  onChange: () => {},
-  code: '{}',
-};
+import { Tooltip } from 'src/common/components/Tooltip';
 
 const StyledConfigEditor = styled(ConfigEditor)`
   &.ace_editor {
     border: 1px solid ${({ theme }) => theme.colors.grayscale.light2};
   }
 `;
 
-function TemplateParamsEditor({ code, language, onChange }) {
-  const [parsedJSON, setParsedJSON] = useState();
+function TemplateParamsEditor({
+  code = '{}',
+  language,
+  onChange = () => {},
+}: {
+  code: string;
+  language: 'yaml' | 'json';
+  onChange: (...args: any) => any;
```

Comment:
how about this?
```suggestion
  onChange: () => void;
```

Reply:
When we are convert `SqlEditor.jsx` to the typescript, we can add function parameter type again. 

https://github.com/apache/superset/blob/d439da2fe00c0a16edfec3fb5354d8e8f4da470b/superset-frontend/src/SqlLab/components/SqlEditor.jsx#L512-L518

Reply:
We should be getting to it soon, and then, yes we'll be able to manage types of props better, ya!